### DISCUSSION TITLE
实验二中测试文件函数名解码问题

### DISCUSSION THREAD

**Discussion by SundayDream00** [arcsysu/SYsU-lang#34] [Q&A] [answered]
版本为更新long long类型之前的，在测试tester/function_test2020/69_last_word_length.sysu.c文件时出现如下提示信息：
```
Traceback (most recent call last):
  File "/home/candy/sysu/bin/sysu-compiler", line 522, in <module>
    exit(main(*sys.argv))
  File "/home/candy/sysu/bin/sysu-compiler", line 497, in main
    return unittest_parser(args.unittest, args.clang,
  File "/home/candy/sysu/bin/sysu-compiler", line 122, in unittest_parser
    ast1 = json.loads(subprocess.run(
  File "/usr/lib/python3.9/json/__init__.py", line 341, in loads
    s = s.decode(detect_encoding(s), 'surrogatepass')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf0 in position 3603: invalid continuation byte
```

parser.y文件中没有设置额外输出，尝试将所指向的函数名`lengthOfLastWord`删去一个字符后就不会再出现该问题，请问这是什么原因呢

**Comment by SundayDream00** [comment]
补充：测试文件`tester/h_functional/094_substr.sysu.c`中也有相关问题，相同报错，parser.y中涉及到的两个自定义函数name输出均为乱码，如图
<img width="450" alt="微信图片_20220429190128" src="https://user-images.githubusercontent.com/98109054/165932466-9123aff9-be1e-44bc-94ef-a0bd31d08095.png">

**Comment by wu-kan** [comment, marked as answer]
说明你的 parser 可能输出了 utf 格式的字符？应该输出为正常的字符流（`char`）才对？检查下你有没有混进去一些全角字符？

**Reply by wu-kan** [reply]
也有可能是你的内存泄露了，访问到了错误的地址
